benchmarks/python/test_matmul.py

```diff
@@ -25,21 +25,73 @@ def load_matmul_problems():
         return list((int(m), int(n), int(k), layout) for m, n, k, layout in reader)
 
 
+@pytest.mark.parametrize("half_reduction", [False, True], ids=["fullred", "halfred"])
+@pytest.mark.parametrize("compile", [False], ids=["eager"])
+@pytest.mark.parametrize("dtype", [torch.float16, torch.bfloat16], ids=["fp16", "bf16"])
 @pytest.mark.parametrize(
-    "config", load_matmul_problems(), ids=lambda val: "_".join(str(v) for v in val)
+    "config", load_matmul_problems(), ids=lambda val: "-".join(str(v) for v in val)
+)
+def test_matmul_baseline_benchmark(
+    benchmark,
+    compile: bool,
+    config: tuple,
+    dtype: torch.dtype,
+    half_reduction: bool,
+    disable_validation: bool,
+    disable_benchmarking: bool,
+):
+    m, n, k, layout = config
+
+    clear_cuda_cache()
+
+    torch.backends.cuda.matmul.allow_fp16_reduced_precision_reduction = half_reduction
+    torch.backends.cuda.matmul.allow_bf16_reduced_precision_reduction = half_reduction
+
+    try:
+        a = torch.randn(m, k, device="cuda", dtype=dtype)
+        b = torch.randn(k, n, device="cuda", dtype=dtype)
+
+        if layout == "NT" or layout == "NN":
+            a = a.as_strided(size=[m, k], stride=[1, m])
+        if layout == "TN" or layout == "NN":
+            b = b.as_strided(size=[k, n], stride=[1, k])
+
+        # NOTE: we never need to validate eager, as it is our baseline
+        if not disable_benchmarking:
```

Comment:
This conditional is not needed. This flag is for nvfuser benchmarks. I'll make a note to rename this. If --benchmark-eager is used we always run this benchmark.